### PR TITLE
Troubleshoot queries

### PR DESCRIPTION
**Pull request by ashwinkumar12345**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/sql/index.md [hunk 1]
```diff
@@ -61,6 +61,8 @@ You can test queries using **Dev Tools** in Kibana (`https://<host>:5601`).
 
 ## Troubleshoot queries
 
+The SQL plugin is stateless, so troubleshooting is mostly focused on why any single query fails.
```
Comment (aetter) [inline docs/sql/index.md:64]: on why a particular query fails.

File: docs/sql/index.md [hunk 2]
```diff
@@ -91,3 +93,56 @@ POST _opendistro/_sql/_explain
...
+    'NOT', 'OR', 'ORDER', 'REGEXP', '*', '/', '%', '+', '-', 'DIV', 'MOD', '=', '>', '<', '!',
+    '|', '&', '^', '.', DOT_ID}",
+  "type": "SyntaxAnalysisException"
+}
+```
+
+To resolve this error:
+
+1. Check if your syntax follows the MySQL grammar.
+1. If your syntax is correct, disable strict query analysis:
```
Comment (aetter) [inline docs/sql/index.md:117]: You can make this entire section a single list. So:

1. Check
1. If

   ```json
   code block here
   ```

1. Run the query again

File: docs/sql/index.md [hunk 3]
```diff
@@ -91,3 +93,56 @@ POST _opendistro/_sql/_explain
...
+  "error": {
+    "reason": "There was internal problem at backend",
+    "details": "When using multiple indices, the mappings must be identical.",
+    "type": "VerificationException"
+  },
+  "status": 503
+}
+```
+
+Make sure the index in your query is not an index pattern and it doesn't have multiple types.
```
Comment (aetter) [inline docs/sql/index.md:146]: is not an index pattern and doesn't have multiple types.

File: docs/sql/index.md [hunk 4]
```diff
@@ -91,3 +93,56 @@ POST _opendistro/_sql/_explain
...
+
+If these workarounds don't work, submit a [Github issue](https://github.com/opendistro-for-elasticsearch/sql/issues).
```
Comment (aetter) [inline docs/sql/index.md:148]: If these steps don't work, submit a GitHub issue [here](url).

File: docs/sql/index.md [hunk 5]
```diff
@@ -91,3 +93,56 @@ POST _opendistro/_sql/_explain
   "size": 50
 }
 ```
+
+### Syntax analysis exception
+
+
+If you get the following "Invalid SQL query" error:
```
Comment (aetter) [inline docs/sql/index.md:100]: You might receive the following error if the plugin can't parse your query:

File: docs/sql/index.md [hunk 6]
```diff
@@ -91,3 +93,56 @@ POST _opendistro/_sql/_explain
...
+    More details: Expecting tokens in {<EOF>, 'AND', 'BETWEEN', 'GROUP', 'HAVING', 'IN', 'IS', 'LIKE', 'LIMIT',
+    'NOT', 'OR', 'ORDER', 'REGEXP', '*', '/', '%', '+', '-', 'DIV', 'MOD', '=', '>', '<', '!',
+    '|', '&', '^', '.', DOT_ID}",
+  "type": "SyntaxAnalysisException"
+}
+```
+
+To resolve this error:
+
+1. Check if your syntax follows the MySQL grammar.
```
Comment (aetter) [inline docs/sql/index.md:116]: Link to the MySQL grammar?

File: docs/sql/index.md (hunk 5 above)
Comment (ashwinkumar12345) [reply docs/sql/index.md:100]: Done.

File: docs/sql/index.md (hunk 4 above)
Comment (ashwinkumar12345) [reply docs/sql/index.md:148]: Done.

File: docs/sql/index.md (hunk 3 above)
Comment (ashwinkumar12345) [reply docs/sql/index.md:146]: Done.

File: docs/sql/index.md (hunk 2 above)
Comment (ashwinkumar12345) [reply docs/sql/index.md:117]: Done.

File: docs/sql/index.md (hunk 1 above)
Comment (ashwinkumar12345) [reply docs/sql/index.md:64]: Done.

File: docs/sql/index.md (hunk 6 above)
Comment (ashwinkumar12345) [reply docs/sql/index.md]: Linked to the SQL docs.

**Pull request closed without merging**